carbon-design-system/carbon
# fix(FileUploaderDropContainer): fix a11y violations, update tests

Closes https://github.com/carbon-design-system/carbon/issues/12498
Refs https://github.com/carbon-design-system/carbon/issues/13039
Refs https://github.com/carbon-design-system/carbon/issues/12725
Refs https://github.com/carbon-design-system/carbon/pull/13061

Refactors `FileUploaderDropContainer` to mimic the changes made in #13061. This creates a `button` element that handles all click/key down behaviors.

I've also added in the `name` prop to `Filename` for better screenreader support. Before, when tabbing to the delete icon, it would just say `delete file` without any indication of the file. 

#### Changelog

**New**

- Refactor `FileUploaderDropContainer` to use a `button`
- `onClick` added to `FileUploaderDropContainer` as well as `composeEventHandlers` to support existing usage of `onClick` by our users

**Changed**

- Updated tests to anticipate new `filename` `aria-label` structure
- Added a `button-reset` to the scss file, and combined the label + "button" classes in the react file

**Removed**

- deprecates `role`, `tabIndex` from `FileUploaderDropContainer` since it now uses a `button` and these are not needed

#### Testing / Reviewing

- Ensure there are no visual regressions
- Ensure all click / keyboard functionality remains the same
- Ensure there are no a11y violations when there are no items uploaded
- Ensure there are no a11y violations when there **are** items uploaded
- Ensure screenreader support is as expected, especially when files have been updated and you are trying to delete them
